raft/status.go

```diff
@@ -47,3 +49,20 @@ func getStatus(r *raft) Status {
 
 	return s
 }
+
+func (s Status) MarshalJSON() ([]byte, error) {
+	j := fmt.Sprintf(`{"id":"%x","term":%d,"vote":"%x","commit":%d, "lead":"%x","raftState":"%s","progress":{`,
```

Comment:
that is why we did not change the original struct. the json is simply for http status. 
for debugging, people should use the struct directly.